frappe/integrations/doctype/gsuite_templates/gsuite_templates.js

```diff
@@ -3,6 +3,50 @@
 
 frappe.ui.form.on('GSuite Templates', {
 	refresh: function(frm) {
-
+		if (frm.is_new()) {
+			// if doc is new, get all options immediately
+			set_available_docs(frm);
+			set_available_folders(frm);
+		}
+	},
+	template_id: function(frm) {
+		if (!frm.is_new()) {
+			// if doc is NOT new, get options when selecting field
+			set_available_docs(frm);
+		}
+	},
+	destination_id: function(frm) {
+		if (!frm.is_new()) {
+			// if doc is NOT new, get options when selecting field
+			set_available_folders(frm);
+		}
 	}
 });
+
+const set_available_docs = (frm) => {
```

Comment:
not a good idea to add global objects, as they can potentially conflict with other declarations that are not namespaced.

add them to the controller object and then call with `frm.trigger('set_available_docs')`